File: code_samples/dart/networking_protocols.dart
// Networking: Protocols
// AI/ML Training Sample

class Protocols {
  String _data = '';
  
  void process(String input) {
    _data = input;
  }
  
  String getData() => _data;
  
  bool validate() => _data.isNotEmpty;
}

void main() {
  final instance = Protocols();
  instance.process('example');
  print('Data: ${instance.getData()}');
  print('Valid: ${instance.validate()}');
}

File: code_samples/scala/networking_protocols.scala
/**
 * Networking: Protocols
 * AI/ML Training Sample
 */

class Protocols {
  private var data: String = ""
  
  def process(input: String): Unit = {
    data = input
  }
  
  def getData: String = data
  
  def validate: Boolean = data.nonEmpty
}

object ProtocolsExample {
  def main(args: Array[String]): Unit = {
    val instance = new Protocols()
    instance.process("example")
    println(s"Data: ${instance.getData}")
    println(s"Valid: ${instance.validate}")
  }
}
